package.json

```diff
@@ -118,6 +118,7 @@
     "ipfs-mfs": "~0.10.2",
     "ipfs-multipart": "~0.1.0",
     "ipfs-repo": "~0.26.5",
+    "ipfs-repo-migrations": "AuHau/js-ipfs-repo-migrations#dev",
```

Comment:
Can't this module just be part of `ipfs-repo`?

Reply:
Well, good question 😅 Maybe bit philosophical and so I am afraid long answer is in need.

**tl;dr: Yes, there is a relationship between the `ipfs-repo-migrations` and `ipfs-repo`, but not necessarily one that would lead to merging these two package.s**

I am not strongly against, but what I have discovered during writing the `ipfs-repo-migrations` is the delicate relationship between migration (& tooling around it) and the model migrated (`ipfs-repo`), where each migration is very much hard-linked to a specific version of the model (that is pretty much desired characteristic). But that relationship directly translates to a link between migration and specific version of the `ipfs-repo` package. 

If you take the main use-case "migrating to the last version of repo", then I don't see any problem there. But since the migration package currently supports actions like "migrate to specific, but not necessarily latest version" or "revert to specific version", these actions would lead to an inconsistent relationship "the current version of the repo vs. what current version of `ipfs-repo` package expects" and hence most probably not expected behavior. 

Another problem is a reversion of the repo. If you want to use some older version of `ipfs-repo` and revert your current repo to that version, you would have to install two versions of `ipfs-repo`. The "latest" one that has all the migrations up to your current repo's version (and hence which can revert the data to desired repo's version) and the actual version of `ipfs-repo` that can work with this version. Of course, having `ipfs-repo-migration` package separate does not solve this per se, as you still have to install a higher version of this package, but it just "makes more sense" in this use case. (At least for me).

That said these problems are, I think, quiet edge-cases. Merging it with `ipfs-repo` would definitely ease the chores around the integration of this tool, but that is definitely not a point that should be considered.

*After this whole write-up and thinking it through, I still believe that these packages should be left not-merged, but I am open to other opinions and/or guidance.*

Reply:
Got it! :)